modules/map/hooks/useNotificationPermission.ts

```diff
@@ -42,29 +45,49 @@ export const useNotificationPermission = () => {
   //  - register device if permissions are granted and set push notifications settings to true
   //  - delete device if permissions are not granted
   //  Now we do only first thing and delete device is called only on sign out.
-  const getPermission = useCallback(async () => {
-    if (Device.isDevice) {
-      // https://rnfirebase.io/messaging/usage#ios---requesting-permissions
-      const authStatus = await messaging().requestPermission()
+  const requestNotificationPermissionAndRegisterDevice = useCallback(async () => {
+    // https://rnfirebase.io/messaging/usage#ios---requesting-permissions
+    if (Platform.OS === 'ios') {
+      // ios simulator does not support notifications
+      if (Device.isDevice) {
+        const authStatus = await messaging().requestPermission()
+        const enabled =
+          authStatus === messaging.AuthorizationStatus.AUTHORIZED ||
+          authStatus === messaging.AuthorizationStatus.PROVISIONAL
+
+        if (enabled) {
+          setPermissionStatus(UnifiedPermissionStatus.GRANTED)
+          await registerDevice()
+          mutateSaveSetting({ pushNotificationsAboutToEnd: true, pushNotificationsToEnd: true })
+        } else {
+          setPermissionStatus(UnifiedPermissionStatus.DENIED)
+        }
+      } else {
+        console.warn('Must use physical device for Push Notifications, skipping.')
+        setPermissionStatus(UnifiedPermissionStatus.DENIED)
+      }
+    }
+
+    // https://rnfirebase.io/messaging/usage#android---requesting-permissions
+    if (Platform.OS === 'android') {
+      await requestNotificationPermissionAndroidAsync()
+
+      // Android doesn't return value of current permission status as messaging().requestPermission() does, so we need to check it manually
+      const authStatus = await messaging().hasPermission()
```

Comment:
This is the main difference from PR #594 where we incorrectly compared the result of `requestNotificationPermissionAndroidAsync` - this does not return actual PermissionStatus, it returns the status of native modal request.
Therefore, we have to call `messaging().hasPermission()` to get the actual PermissionStatus.

The whole hook needs refactoring together with Permissions Screen, so a little bit of duplicate code here should not be a problem.